platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,208 @@
+
+##### Anybody that is trying to ship code for the VA.gov front end
+
+If invalid content is breaking the production build, and therefore blocking the production deploy, then no team on any contract is able to ship their frontend code to VA.gov.
+
+### High Level Design
+
+Once the `content-build` repo is successfully extracted from `vets-website`, the Metalsmith static content build pipeline will be deleted from `vets-website`. Webpack will handle the application build in `vets-website`, and Metalsmith will handle the static content build in the `content-build` repo. This will create two separate pipelines for built code to be deployed to S3, which will prevent failures in one from blocking the other.
+
+The validation of the static content coming from the CMS will occur during the Metalsmith build.
+
+## Specifics
+
+### Detailed Design
+
+The content validation will happen in a daily scheduled job.
+
+### Code Location
+
+The content validation script and scheduled job config will live in the [new `content-build` repo](https://github.com/department-of-veterans-affairs/content-build).
+
+### Testing Plan
+
+1. Configure a scheduled job to run a script once per workday
+2. Write an initial script that logs when the script is run
+3. Once the scheduled job is running at the correct time, replace the initial script with the actual script that runs the content validation
+
+### Logging
+
+Currently, invalid content is logged during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
+
+### Debugging
+
+##### Broken links
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html) to have a broken link
+3. Run the build again to confirm that it is failing
+
+##### Accessibility errors
+
+1. Confirm that the content build is passing
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html) to have an accessibility error
+3. Run the build again to confirm that it is failing
+
+### Caveats
+
+- Our [accessibility check](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/assets/js/execute-axe-check.js#L8) only renders a banner on the preview server.
+
+### Security Concerns
+
+There are no new security concerns with a separated build process.
+
+### Privacy Concerns
+
+There are no new privacy concerns with a separated build process.
+
+### Open Questions and Risks
+
+#### Risks
+
+- Neither the current content validation nor the proposed content validation will **prevent** invalid content from making it to production. But this proposal would prevent unrelated teams from being blocked by invalid content.
```

Comment:
This might be me being picky, but I'm not sure this is 100% true.  When you say "production" here, are you talking about the live va.gov site, or do you mean the CMS production server?  We have had incidents of bad content breaking the content build, which prevents new content from going to prod on va.gov